### PR TITLE
143 otp encryption

### PR DESCRIPTION
**Pull request by SreeRanjani14**

### CODE REVIEW

File: src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs [hunk 1]
```diff
@@ -199,6 +199,8 @@ public AcceptedResult SetTransactionId(AuthOnInitRequest request)
                 return StatusCode(StatusCodes.Status400BadRequest, error);
             var requestId = gatewayAuthConfirmRequestRepresentation.requestId;
             var cmSuffix = gatewayAuthConfirmRequestRepresentation.cmSuffix;
+            
+            gatewayAuthConfirmRequestRepresentation.credential.authCode = UserAuthService.GetDecodedOtp(authConfirmRequest);
```
Comment (MeghnaThoughtworks) [inline src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs:203]: Can we push this line to UserAuthService itself? The Service sole purpose was to return the correct gatewayRepresentations. Controllers should not handle decoding of the otp or any service related work.

File: src/In.ProjectEKA.HipService/UserAuth/UserAuthService.cs [hunk 2]
```diff
@@ -87,6 +89,13 @@ private static bool IsPresentInMap(string healthId)
             return UserAuthMap.HealthIdToTransactionId.ContainsKey(healthId);
         }
 
+        public static string GetDecodedOtp(AuthConfirmRequest authConfirmRequest)
```
Comment (MeghnaThoughtworks) [inline src/In.ProjectEKA.HipService/UserAuth/UserAuthService.cs:92]: make this private as well

**Review by MeghnaThoughtworks — CHANGES_REQUESTED** [review]

File: src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs (hunk 1 above)
Comment (Nazeer-debug) [reply src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs:203]: We don't think so, anyhow we need to call that method even after moving that method to UserAuthService, UserAuthSerice is handling the response related things

Comment (MeghnaThoughtworks) [reply src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs:205]: authCode is a part gatewayResponse, isn't it?

Comment (MeghnaThoughtworks) [reply src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs:205]: I'm not talking about moving a method only the method call.

Comment (MeghnaThoughtworks) [reply src/In.ProjectEKA.HipService/UserAuth/UserAuthController.cs:205]: You are doing a service related work as in changing the authCode in controller? do you get that is my only reason. you can call that method inside the userAuthService.AuthConfirmResponse(authConfirmRequest) only.

**Review by MeghnaThoughtworks — CHANGES_REQUESTED** [review]

**Pull request closed without merging**